src/entities/Board.js

```diff
@@ -0,0 +1,32 @@
+import { GameObjects, Plugins } from "phaser";
```

Comment:
Why plugins?

Reply:
I think the `pluginManager.registerGameObject` bit makes it so that you can say `this.add.board` over in `MainScene`. It seems like it will also automatically partially apply the `scene` parameter in the `Board` constructor. And I guess it gives us a spot to call `displayList.add`, which is what ultimately gets rendered I think.

But other than that I'm not really sure what else it does and to me it doesn't feel super worth it.

I think I like how `Player.js` (meant to rename to `Piece.js`) turned out better where it doesn't extend from anything and is more just a container for whatever Phaser stuff we might need to do. It felt weird to say "the piece _is a_ star" by extending from `GameObjects.Star`. How it is now I think it's more like "the piece _has a_ star".

Reply:
But I got the Plugin stuff from this example:

https://www.phaser.io/examples/v3/view/plugins/custom-game-object

Feels like Phaser is VERY configuration > convention / do whatever you want. But that might be because it seems like it gets massively refactored in each version and there's no overarching guide to how to use it for anything of scale so you just get these smattering of out of context examples written in different styles.